zonemaster/zonemaster-gui
# Make it possible to look for a domain history without running a test

## Purpose

This PR is about providing an easy way to fetch previous results for a given domain.
A new page has been created that provides a form similar to the domain check's form to look for a test's history. The history list is displayed on the page reusing the modal (from the result page).

Two new routes are created:
* `/history` that displays the page with the history form
* `/history/<domain>` that populates the form and queries for the history of \<domain\>.

![zmgui-history2](https://user-images.githubusercontent.com/70589067/189932460-875eb629-b9f1-46b0-bfc4-a0d880b05376.png)
![zmgui-history3](https://user-images.githubusercontent.com/70589067/189932466-bdb65228-14b5-410f-b6b0-a3b927d84f8d.png)

## Context

Addresses #309 

## Changes

* New route `/history`
* New route `/history/<domain>`
* New "History" page with a form
* The `sanitizeDomain()` method from the form component has been moved to an external file to make it available to the project (expand this method to convert the domain to lower case)

## How to test this PR

Open the History page. Here is a list of things that can be tested:
* get the history for a \<domain\> using the form
* access the history from the URL directly `/history/<domain>`
* open a result from the history page
* switch language from the history page
* check that the modal window is not altered (from the result page)
* play with the browser history

---

## Old proposal

### 1. Display a secondary button
<details><summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/70589067/179566092-699e06ae-7f48-49f8-b1a7-4927c77dfe49.png)
![image](https://user-images.githubusercontent.com/70589067/179566034-e6c9282c-f71f-4dc9-9d39-50f68c56093e.png)

</details>
